### PR TITLE
feat: Allow finding commits by prefix

### PR DESCRIPTION
**Pull request by hego-fdc**

### CODE REVIEW

File: pkg/valid/validations.go [hunk 1]
```diff
@@ -22,18 +22,19 @@ import (
...
-	TeamNameRegExp = AppNameRegExp
-	EnvNameRegExp  = AppNameRegExp
-	CommitIDRegExp = `^[0-9a-fA-F]{40}$`
+	MaxAppNameLen        = 39
+	AppNameRegExp        = `\A[a-z0-9]+(?:-[a-z0-9]+)*\z`
+	TeamNameRegExp       = AppNameRegExp
+	EnvNameRegExp        = AppNameRegExp
+	SHA1CommitIDLength   = 40
+	commitIDPrefixRegExp = `^[0-9a-fA-F]*$`
 )
```
Comment (khaled-ismaeel) [inline pkg/valid/validations.go:31]: Perhaps this is a better way for writing this regex.

```suggestion
	commitIDPrefixRegExp = `^[0-9a-fA-F]{1,40}$`
```

File: pkg/valid/validations.go [hunk 2]
```diff
@@ -22,18 +22,19 @@ import (
...
-	AppNameRegExp  = `\A[a-z0-9]+(?:-[a-z0-9]+)*\z`
-	TeamNameRegExp = AppNameRegExp
-	EnvNameRegExp  = AppNameRegExp
-	CommitIDRegExp = `^[0-9a-fA-F]{40}$`
+	MaxAppNameLen        = 39
+	AppNameRegExp        = `\A[a-z0-9]+(?:-[a-z0-9]+)*\z`
+	TeamNameRegExp       = AppNameRegExp
+	EnvNameRegExp        = AppNameRegExp
+	SHA1CommitIDLength   = 40
+	commitIDPrefixRegExp = `^[0-9a-fA-F]*$`
```
Comment (khaled-ismaeel) [inline pkg/valid/validations.go:30]: Perhaps this is a better way for writing this regex.

```suggestion
	commitIDPrefixRegExp = `^[0-9a-fA-F]{1,40}$`
```

File: services/cd-service/pkg/service/git_test.go [hunk 3]
```diff
@@ -393,6 +395,32 @@ func TestGetCommitInfo(t *testing.T) {
...
+				Events: []*api.Event{
+					{
+						CreatedAt: timestamppb.New(fixedTime()),
+						EventType: &api.Event_CreateReleaseEvent{
+							CreateReleaseEvent: &api.CreateReleaseEvent{},
+						},
+					},
+				},
+			},
+		},
```
Comment (khaled-ismaeel) [inline services/cd-service/pkg/service/git_test.go:423]: Perhaps we need to add test for the case when the commit ID length is less than 7 and greater than 40?

File: pkg/valid/validations.go (hunk 2 above)
Comment (hego-fdc) [reply pkg/valid/validations.go:30]: I think it's better to check the length separately, as this gives you the ability to generate more precise error messages.

File: services/cd-service/pkg/service/git_test.go (hunk 3 above)
Comment (hego-fdc) [reply services/cd-service/pkg/service/git_test.go:423]: I don't think that those are especially important cases. Even if we would accidentally give out a commit if the length is <7 , there's no harm in it.

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**